Assets/LeapMotion/Scripts/LeapServiceProvider.cs

```diff
@@ -177,16 +203,23 @@ public class LeapServiceProvider : LeapProvider {
       }
 #endif
 
+      if (_updateHandInPrecull!=_prevUpdateHandInPrecull && !_updateHandInPrecull) {
+        _transformArray[0] = Matrix4x4.identity;
+        _transformArray[1] = Matrix4x4.identity;
+        Shader.SetGlobalMatrixArray("_handTransforms", _transformArray);
+      }
+      _prevUpdateHandInPrecull = _updateHandInPrecull;
+
       _fixedOffset.Update(Time.time - Time.fixedTime, Time.deltaTime);
 
       if (_useInterpolation) {
 #if UNITY_ANDROID
-        Int64 time = leap_controller_.Now() - (_interpolationDelay+16) * 1000;
+        Int64 time = leap_controller_.Now() - ((_interpolationDelay+16) * 1000) + (_updateHandInPrecull ? (long)(Time.smoothDeltaTime * S_TO_NS / Time.timeScale) : 0);
         leap_controller_.GetInterpolatedFrame(_untransformedUpdateFrame, time);
 #else
-        _smoothedTrackingLatency.value = Mathf.Min(_smoothedTrackingLatency.value, 25000f);
+        _smoothedTrackingLatency.value = Mathf.Min(_smoothedTrackingLatency.value, 30000f);
         _smoothedTrackingLatency.Update((float)(leap_controller_.Now() - leap_controller_.FrameTimestamp()), Time.deltaTime);
-        leap_controller_.GetInterpolatedFrame(_untransformedUpdateFrame, leap_controller_.Now() - (long)_smoothedTrackingLatency.value - (_interpolationDelay * 1000));
+        leap_controller_.GetInterpolatedFrame(_untransformedUpdateFrame, leap_controller_.Now() - (long)_smoothedTrackingLatency.value - (_interpolationDelay * 1000) + (_updateHandInPrecull ? (long)(Time.smoothDeltaTime * S_TO_NS / Time.timeScale) : 0));
```

Comment:
This offset calculation is done in many places, probably should be put in its own method to reduce duplication.